src/main/java/seedu/address/logic/parser/ShowCommandParser.java

```diff
@@ -47,7 +48,7 @@ public ShowCommand parseByIndex(ArgumentMultimap argMultimap, Path savePath) thr
         try {
             index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ShowCommand.MESSAGE_USAGE));
+            throw new ParseException(MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
```

Comment:
Perhaps we need a more descriptive error message than "The student index provided is invalid".
I suggest we split the error message into the following cases:

* Inputs that contain characters that are not digits: "Index is not a positive unsigned integer."
* Inputs that only contain digits
    * Those that `parseInt` properly: 
        * If 0: "Index is not a positive unsigned integer."
        * If greater than student list size: "Index is too large"
        * Else: Executes properly
    * Doesn't `parseInt`: Above `MAX_INT`: "Index is too large"

Would be nice if you could rewrite the `parseIndex` function in `ParserUtil.java` to implement these (except perhaps the greater than student list size one)